app/middleware/concatenation.ts

```diff
@@ -0,0 +1,66 @@
+import { NextFunction, Response } from 'express';
+import { ParameterParseError, parseBoolean } from '../frontends/ogc-coverages/util/parameter-parsing';
+import HarmonyRequest from '../models/harmony-request';
+import { RequestValidationError } from '../util/errors';
+import { keysToLowerCase } from '../util/object';
+
+/**
+ * Middleware to determine whether the request should concatenate results. Called prior
+ * to choosing the service that will be used.
+ *
+ * @param req - The client request, containing an operation
+ * @param res - The client response
+ * @param next - The next function in the middleware chain
+ */
+export function preServiceConcatenationHandler(
+  req: HarmonyRequest, _res: Response, next: NextFunction,
+): void {
+  const query = keysToLowerCase(req.query);
```

Comment:
Realizing we do this lowercasing everywhere. It's probably begging for a move to a middleware or something unless we ever care about keys in a case-sensitive way.

Reply:
(Not necessary for this PR)